vassal-app/src/main/java/VASSAL/build/module/map/PieceMover.java

```diff
@@ -806,6 +809,58 @@ public Command movePieces(Map map, Point p) {
       // Any piece we successfully moved, make sure is now considered a "selected piece" (i.e. a member of KeyBuffer)
       for (final GamePiece piece : draggedPieces) {
         KeyBuffer.getBuffer().add(piece);
+
```

Comment:
We will probably need some selection criteria here. Not all Cargo may be allows on all mats. The Mat should have an optional PropertyExpression to limit the cargo that is allowed on it. Only my pieces on my mat for example.